irr/src/os.h

```diff
@@ -10,6 +10,21 @@
 #include "ILogger.h"
 #include "ITimer.h"
 
+// CODE_UNREACHABLE(): Invokes undefined behavior for unreachable code optimization
+#if defined(__cpp_lib_unreachable)
+#define CODE_UNREACHABLE() std::unreachable()
+#elif defined(__has_builtin)
+#if __has_builtin(__builtin_unreachable)
+#define CODE_UNREACHABLE() __builtin_unreachable()
+#endif
+#elif defined(_MSC_VER)
+#define CODE_UNREACHABLE() __assume(false)
+#endif
+
+#ifndef CODE_UNREACHABLE
+#define CODE_UNREACHABLE() std::abort()
```

Comment:
I guess I should have said that in absence of a working implementation I think this should just expand to nothing

Reply:
That would be ideal, but compilers like MinGW would not know a warning is unnecessary. I will go head and remove the abort code anyways.

Reply:
Done in 943ea3e57

Reply:
Not sure what you mean. MinGW is gcc or clang which supports `__builtin_unreachable`